src/Symfony/Component/ObjectMapper/Tests/Fixtures/MapStruct/MapStructMapperMetadataFactory.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\ObjectMapper\Tests\Fixtures\MapStruct;
+
+use Symfony\Component\ObjectMapper\Attributes\Map;
+use Symfony\Component\ObjectMapper\MapperMetadataFactoryInterface;
+
+/**
+ * @internal
+ *
+ * @author Antoine Bluchet <soyuka@gmail.com>
+ */
+final class MapStructMapperMetadataFactory implements MapperMetadataFactoryInterface
```

Comment:
@maxbeckers @rogeriolino 

here's a prototype that works like MapStruct, I didn't change much but indeed the `target` was missing from the attribute, not sure if Symfony wants to have two different approaches for mapping as it may be confusing so for now this is inside the fixtures and needs to be cleaned if we do want both approaches.

Reply:
@soyuka thanks for your work on this great new component. I'll try to find tomorrow the time to check this changes out and give you some feedback :)
Yes to come with one component supporting two different approaches can be confusing ... but maybe both cases have their use cases.